osbenchmark/utils/opts.py

```diff
@@ -125,7 +125,11 @@ class StoreKeyPairAsDict(argparse.Action):
     """
     def __call__(self, parser, namespace, values, option_string=None):
         custom_dict = {}
-        for kv in values:
+
+        # If values contains spaces, user provided 2+ key value pairs
+        kv_pairs = values[0].split(" ")
+
+        for kv in kv_pairs:
```

Comment:
fIs this the case where the user provides multiple entries in a single arg within shell quotes?
```
--number-of-docs "x:3 y:4 z:8"
```
In that case, `nargs` should be 1, not +.
To handle both cases with +, this code should be activated only when len(values) is 1.  In other cases, this should fall back to the earlier mode of processing.

Reply:
Got it. Will implement it to where it handles both use cases.